### PR TITLE
Fail on superfluous parameters

### PR DESCRIPTION
**Pull request by rauchy**
When running a `ParameterizedQuery` with a schema (as in the case of saved queries), `is_safe` should fail if there are any superfluous parameters which are not specified in the schema.

### CODE REVIEW

File: redash/utils/parameterized_query.py [hunk 1]
```diff
@@ -96,10 +96,13 @@ def apply(self, parameters):
         return self
 
     def _valid(self, name, value):
+        if not self.schema:
+            return True
```
Comment (arikfr) [inline redash/utils/parameterized_query.py]: Why is this needed?

Comment (arikfr) [reply redash/utils/parameterized_query.py]: For the non-textless API?

Comment (rauchy) [reply redash/utils/parameterized_query.py]: Precisely.

Comment (arikfr) [reply redash/utils/parameterized_query.py]: Feels like it might bite us in the future, but can't think of any reason why.
